tutorials/misc/plot_ecog.py

```diff
@@ -9,71 +11,114 @@
 #
 # License: BSD (3-clause)
 
+import pandas as pd
 import numpy as np
 import matplotlib.pyplot as plt
-from scipy.io import loadmat
 
 import mne
 from mne.viz import plot_alignment, snapshot_brain_montage
 
 print(__doc__)
 
+# paths to mne datasets - sample ECoG and FreeSurfer subject
+misc_path = mne.datasets.misc.data_path()
+sample_path = mne.datasets.sample.data_path()
+
 ###############################################################################
 # Let's load some ECoG electrode locations and names, and turn them into
 # a :class:`mne.channels.DigMontage` class.
+# First, use pandas to read in the .tsv file.
+
+# In this tutorial, the electrode coordinates are assumed to be in meters
+elec_df = pd.read_csv(misc_path + '/ecog/sample_ecog_electrodes.tsv',
+                      sep='\t', header=0, index_col=None)
+ch_names = elec_df['name'].tolist()
+ch_coords = elec_df[['x', 'y', 'z']].to_numpy(dtype=float)
+ch_pos = dict(zip(ch_names, ch_coords))
+
+# Now we make a :class:`mne.channels.DigMontage` stating that the ECoG
+# contacts are in head coordinate system (although they are in MRI). This is
+# compensated below by the fact that we do not specify a trans file so the
+# Head<->MRI transform is the identity.
 
-mat = loadmat(mne.datasets.misc.data_path() + '/ecog/sample_ecog.mat')
-ch_names = mat['ch_names'].tolist()
-elec = mat['elec']  # electrode positions given in meters
-# Now we make a montage stating that the sEEG contacts are in head
-# coordinate system (although they are in MRI). This is compensated
-# by the fact that below we do not specicty a trans file so the Head<->MRI
-# transform is the identity.
-montage = mne.channels.make_dig_montage(ch_pos=dict(zip(ch_names, elec)),
-                                        coord_frame='head')
+montage = mne.channels.make_dig_montage(ch_pos, coord_frame='head')
 print('Created %s channel positions' % len(ch_names))
 
 ###############################################################################
-# Now that we have our electrode positions in MRI coordinates, we can create
-# our measurement info structure.
+# Now that we have our montage, we can load in our corresponding
+# time-series data and set the montage to the raw data.
+
+# first we'll load in the sample dataset
+raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
+
+# drop bad channels
+raw.info['bads'].extend([ch for ch in raw.ch_names if ch not in ch_names])
+raw.load_data()
+raw.drop_channels(raw.info['bads'])
+
+# attach montage
+raw.set_montage(montage)
 
-info = mne.create_info(ch_names, 1000., 'ecog').set_montage(montage)
+###############################################################################
+# We then compute the signal power in certain frequency bands (e.g. 30-90 Hz).
+# We compute the power in gamma and alpha bands.
+gamma_power = np.sum(raw.copy().filter(30, 90).get_data() ** 2, axis=1)
+alpha_power = np.sum(raw.copy().filter(8, 12).get_data() ** 2, axis=1)
```

Comment:
this comes too early; the electrode location plot should come first in my opinion.